core/lib/spree/preferences/preferable_class_methods.rb

```diff
@@ -9,14 +11,18 @@ module PreferableClassMethods
       password
       string
       text
+      encrypted_string
     )
 
     def defined_preferences
       []
     end
 
     def preference(name, type, options = {})
-      options.assert_valid_keys(:default)
+      options.assert_valid_keys(:default, :encryption_key)
+      preference_encryptor = preference_encryptor(options) if type == :encrypted_string
+
+      options[:default] = preference_encryptor.encrypt(options[:default]) if type == :encrypted_string
```

Comment:
These postfixed `if` are the same for the 2 lines above, what about grouping them together in a single `if`?

Reply:
Usually, I don't like multiline if, but I've refactored the code using a multiline if and it's not bad, thanks for the feedback.